course-material/intro-to-sql/combining-data-from-multiple-tables.md

```diff
@@ -1,14 +1,305 @@
+
+```tip
+The above query can also be written without the `INNER JOIN` clause! Another way to write the query is as follows:
+
+~~~python
+SELECT a.account_id, c.card_id, d.district_id, l.disp_id
+FROM s1.account as a, s1.card as c, s1.district as d, s1.link as l
+WHERE a.district_id = d.district_id AND
+      l.account_id = a.account_id AND
+      c.disp_id = l.disp_id;
+~~~
+```
+
+</details>
+<!-- #endregion -->
+
+### Question 2 (Medium):
+If tables `links` and `cards` did not have matching rows that are related to each other, what kind of join would you use to combine these tables? If you were to use this join, and then join the resulting table with the merged table of `account` and `district`, how would the final output differ from Question 1?
+
+<!-- #region -->
+<details>
+
+<summary>Answers</summary>
+
+A `FULL OUTER JOIN` would be used to combine the `links` and `cards` tables if we want to keep both the rows that can be matched and the unpaired rows. This is because, in this case, the tables `links` and `cards` do not have matching rows that are related to each other. Therefore, we need to account for the unpaired rows in the result set. 
+
+The `FULL OUTER JOIN` will return all rows from both tables, and if there are no matches, the columns from the other table will be filled with `NULL` values. This is in contrast to the `INNER JOIN`, which only returns rows that have matching values in both tables and, hence, no `NULL` values.
+
+Code to implement the `FULL OUTER JOIN` is shown below:
+
+```{code-cell} ipython3
+%%sql
+SELECT a.account_id, c.card_id, d.district_id, l.disp_id
+FROM s1.link AS l
+FULL OUTER JOIN s1.card AS c 
+ON l.disp_id = c.disp_id 
+INNER JOIN s1.account AS a 
+ON a.account_id = l.account_id 
+INNER JOIN s1.district AS d 
+ON d.district_id = a.district_id;
+```
+
+Therefore, the final output will include all rows from the `links` and `cards` tables, as well as the rows from the `account` and `district` tables that have matching values in the `links` and `cards` tables. Because the `links` table has the highest number of rows, 5369, all of them will be included in the final output. The rows from the `account` and `district` tables that do not have matching values in the `links` and `cards` tables will not be included in the final output. Since a `FULL OUTER JOIN` is used first, the final output will include all rows from the `links` and `cards` tables, even if there are no matching values in the `account` and `district` tables.
+
+</details>
+<!-- #endregion -->
+
+### Question 3 (Medium):
+Create a junction table, `account_district`.
+
+<b>Hint</b> `COUNT()` is a aggregating function in SQL (more on aggregation later!). Try experimenting with `COUNT()` in your `SELECT` clause to see if you can find the correct count.
+
+<!-- #region -->
+<details>
+<summary>Answers</summary>
+
+```{code-cell} ipython3
+%%sql
+SELECT 
```

Comment:
This suggestion feels disjointed from the hint and the question.

Select won't create a table